### ISSUE TITLE
bug: radio button can be deselected with a mouse click

### ISSUE THREAD

**Issue by scomea** [microsoft/fast#3887]
**Describe the bug; what happened?**
A radio button can be selected via keyboard by pressing space when it has focus, but pressing space again does nothing, pressing enter does nothing.  Clicking a selected radio button with the mouse deselects it.

**What are the steps to reproduce the issue?**
Go to https://explore.fast.design/components/fast-radio
Tab to the sample radio button
hit enter - nothing happens
hit space - button selected
hit space - nothing happens
click on the button - button deselected

**What behavior did you expect?**
radio button cannot be deselected by a mouse click

**Comment by chrisdholt** [comment]
@scomea, @eisenbergeffect I don't think the expectation here is correct - once a radio button within a radiogroup is selected, I don't think it can be unselected. I think the issue present from the repro steps above would be that clicking on the button of a selected item should be a noop.

Link: https://w3c.github.io/aria-practices/#radiobutton

**Comment by SethDonohue** [comment]
Yeah we just discussed, the button click is wrong, keyboarding works.

**Comment by scomea** [comment]
Updated title and description to show that the issue is with the mouse click to deselect.
